src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -138,6 +144,40 @@ Kokkos::View<SizeType *, typename ViewType::device_type> sortObjects(
 }
 #endif
 
+#if defined(KOKKOS_ENABLE_SYCL)
+// NOTE returns the permutation indices **and** sorts the input view
+template <typename ViewType, class SizeType = unsigned int>
+Kokkos::View<SizeType *, typename ViewType::device_type>
+sortObjects(Kokkos::Experimental::SYCL const &space, ViewType &view)
+{
+  int const n = view.extent(0);
+
+  using ValueType = typename ViewType::value_type;
+  static_assert(std::is_same<std::decay_t<decltype(space)>,
+                             typename ViewType::execution_space>::value,
+                "");
+
+  Kokkos::View<SizeType *, typename ViewType::device_type> permute(
+      Kokkos::ViewAllocateWithoutInitializing("ArborX::Sorting::permutation"),
+      n);
+  ArborX::iota(space, permute);
+
+  sycl::buffer<typename ViewType::value_type, 1> view_buffer{view.data(),
+                                                             sycl::range<1>{n}};
+  sycl::buffer<SizeType, 1> permute_buffer{permute.data(), sycl::range<1>{n}};
+  auto keys_begin = oneapi::dpl::begin(view_buffer);
+  auto vals_begin = oneapi::dpl::begin(permute_buffer);
+
+  auto zipped_begin = oneapi::dpl::make_zip_iterator(keys_begin, vals_begin);
+  oneapi::dpl::execution::device_policy policy;
```

Comment:
Hmm. I guess if one wants to use `std::sort`, this is one way to do it. Would want to check if it's faster to first construct the permutation indices, and then to apply the permutation.